Orphans in non showcontent categories

SmileyChris opened the pull request:
# Description
Currently, orphans (newsfragments without an issue number) don't display correctly in a non "showcontent" category, such as the "misc" category.

These fragments don't have an issue number, but if they were important enough to have a fragment, I still think their content should then be shown.

# Checklist
<!-- add a "X" inside the brackets to confirm -->
* [x] Make sure changes are covered by existing or new tests.
* [x] For at least one Python version, make sure local test run is green.
* [x] Create a file in `src/towncrier/newsfragments/`. Describe your
  change and include important information. Your change will be included in the public release notes.
* [x] Make sure all GitHub Actions checks are green (they are automatically checking all of the above).
* [x] Ensure documentation is still up-to-date.

SmileyChris committed:
Always content for orphans, since they don't have a concept of tickets

SmileyChris committed:
Add tests for orphans in non showcontent categories

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

SmileyChris committed:
Add newsfragment

SmileyChris committed:
Add a note in the configuration docs about the change to orphan fragments in non-showcontent categories

SmileyChris committed:
Refactor rendering of title via `config.title_format` (#610)

* Refactor rendering of title via config.title_format

* Add newsfragment

* Config docs

* Fix restructuredtext formatting error

adiroiban commented on docs/configuration.rst:
```diff
@@ -212,6 +212,8 @@ These may include the following optional keys:
 ``showcontent``
     A boolean value indicating whether the fragment contents should be included in the news file.
 
+    Note that orphan fragments (those without an issue number) will still show their content.
```
Maybe have it as an RST admonition  ... but below the `true by default` line

```suggestion
.. note::

    Orphan fragments (those without an issue number) always have their content included.
    If a fragment was created, it means that information is important for end users.
```

adiroiban commented on docs/configuration.rst:
```diff
@@ -253,6 +255,8 @@ Each table within this array has the following mandatory keys:
 ``showcontent``
     A boolean value indicating whether the fragment contents should be included in the news file.
 
+    Note that orphan fragments (those without an issue number) will still show their content.
```
same as the other comment.
Have it as a dedicated `RST note`

adiroiban commented on src/towncrier/templates/default.md:
```diff
@@ -15,7 +15,6 @@
 {% for category, val in definitions.items() if category in sections[section] %}
 ### {{ definitions[category]['name'] }}
 
-{% if definitions[category]['showcontent'] %}
```
I think that we should add an `612.removal` fragment, informat that when `showcontent = False` the template will no longer receive the fragment content.

This is a note, just in case someone was missusing this feature with a custom template.

SmileyChris replied:
I think there's some confusion here. The existing behaviour is already that fragment content is not sent to the template (https://github.com/twisted/towncrier/pull/612/files#diff-36db088409efe07783d49792a18cec6ae3778ace338bf1cd08daa8a3fce29e93L166). This change just means that orphan fragment contents **are** sent.

adiroiban replied:
True. My bad. I misinterpreted the changes in the code.

adiroiban reviewed (approved):
Thanks. Looks good.

My only major comment is to add a `.removal` release note informing that the Jinja2 template will no longer receive the fragment content, for `showcontent = False` categories.

SmileyChris committed:
Use rst admonitions for the showcontent notes

SmileyChris committed:
Refactor issue_key function to sort issues in a human-friendly way (#608)

* Refactor issue_key function to sort issues in a human-friendly way

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Rename newsfragment

* Small improvement to test to show how text with numeric issues are sorted

* Update src/towncrier/_builder.py docstring grammar

Co-authored-by: Adi Roiban <adiroiban@gmail.com>

* clarify new behaviour in newsfragment

* Add some docstrings/comments to tests

* linelength fix

* Clarify news fragments vs tickets

Co-authored-by: Adi Roiban <adiroiban@gmail.com>

* Consistent use of "issue" rather than "ticket"

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* typo

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Adi Roiban <adiroiban@gmail.com>

SmileyChris committed:
variable now called issue instead of ticket

adiroiban reviewed (approved):
All good. Thanks.

Sorry for the previous review that was mediocre.

SmileyChris committed:
Merge remote-tracking branch 'upstream/trunk' into non-showcontent-orphans
